### DISCUSSION TITLE
Packaging modular flet app

### DISCUSSION THREAD

**Discussion by farshadz1997** [flet-dev/flet#1277] [Q&A]
### Question

Hello,

I want to create an executable for my application and it's modular, I don't have any idea how to pack modules with along together.
this is my directory scheme:

```
.
├── main.pyw
├── requirements.txt
├── assets/
│   └── fonts/
│       └── *.otf
└── src/
    ├── core/
    │   ├── *.py
    │   └── something.txt
    └── ui/
        └── *.py
```
In ui folder there are scripts for creating Flet controls and layout. 

requirements.txt:

```requirements.txt
flet==0.3.2
certifi
chardet
idna
requests
selenium==4.4.3
urllib3
ipapi
func_timeout
random-word==1.0.7
pyyaml
notifiers
python-slugify
```

if you want to know more about the project [check this](https://github.com/farshadz1997/Microsoft-Rewards-bot-GUI-V2).

Thanks in advance for your help.

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by lawchin** [comment]
Yeah...... facing the same problem! webapp not working after adding Pyrebase4. Can't find any tutorial related to this matter not in youtube not in google not stackoverflow..... I guess its still too new. but yeah still looking for answer!! please anyone

**Comment by FeodorFitsner** [comment]
@lawchin I don't think your issue is related to this one. If you are publishing Flet app to a static website it could be some incompatibility issue with Pyrebase4 and Pyodide. Please create a separate issue with a code to reproduce, error message, etc. and we'll start from there.

**Reply by lawchin** [reply]
owh! sorry I didn't know that. I though it what the modules problem. Let me have a look what Pyodide does. I hope I can understand it.

**Comment by farshadz1997** [comment]
@FeodorFitsner I admire your work and other contributors.

Could you please help me with this if you know how to do this?

**Reply by FeodorFitsner** [reply]
Have you tried https://flet.dev/docs/guides/python/packaging-desktop-app? Did you get any errors?

**Reply by farshadz1997** [reply]
Yes, there's a problem with ```random-word``` module.

![image](https://user-images.githubusercontent.com/60227955/232160621-c89290aa-6a75-4ec3-8840-d949a0e69ab7.png)

when I exclude this module from the script everything works fine.  I thought maybe I could use Pyinstaller arg ```--collect-all``` but Flet CLI doesn't accept any other argument. Do you have any idea how can I address this issue?

**Reply by thetrotfreak** [reply]
@farshadz1997, why not use `PyInstaller` directly? That'll give you more control over **what and how** to generate. Your app builds and runs just fine that way. I tried.

**Reply by farshadz1997** [reply]
I tried multiple times but didn't go well, how did you use Pyinstaller for Felt?

**Comment by farshadz1997** [comment]
I managed to fix it by adding the whole library into codes.
